### ISSUE TITLE
Repo: Build failures on vitest.config.base

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#11066]
### Suggestion

Example from https://github.com/typescript-eslint/typescript-eslint/actions/runs/14445209812/job/40504286765:

```plaintext
Vitest caught 1 unhandled error during the test run.
This might cause false positive tests. Resolve unhandled errors to make sure your tests are not affected.
⎯⎯⎯ Unhandled Source Error ⎯⎯⎯
TypeCheckError: Output file '/home/runner/work/typescript-eslint/typescript-eslint/dist/out-tsc/root/vitest/vitest.config.base.d.mts' has not been built from source file '/home/runner/work/typescript-eslint/typescript-eslint/vitest.config.base.mts'.
 ❯ vitest.config.mts:4:34
```

### Additional Info

As noted by @kirkwaiblinger, this started around https://github.com/typescript-eslint/typescript-eslint/commit/af6ace75d8d348309994afd50509a3abad5c9167. 

💖
